### DISCUSSION TITLE
The problem of using LovyanGFX to broadcast an image from an esp32-cam camera to a gc9a01 tft screen

### DISCUSSION THREAD

**Discussion by marlen-subaru** [lovyan03/LovyanGFX#659] [Q&A]
Good afternoon, dear authors. Please help me. I have a project based on esp 32-cam and gc9a01 with the TFT_ips library, where the image is broadcast from the camera to the tft screen. But, the frame rate, the speed, is less than what you have here. I recently found your job. And I wanted to use your LovyanGFX library. But I couldn't, I have a lot of mistakes, solving which new ones appear. Help me write the code to broadcast the image from the esp32-cam camera to the gc9a01 tft screen. I am also interested in the possibilities of using resize in this code.

[edit] email removed (spam policy)

**Comment by tobozo** [comment]
hi,

support on this forum is limited to LovyanGFX e.g. you can get help to tweak the panel config or identify a bug.

however to get such help you need to provide more information:

- what is `TFT_ips` library?
- what is the esp32 model and what type of psram does it use?

coding in C++ is all about solving the next error until it works, and if you use the issue/discussion search on this repository, you'll find that most problems already have a solution

**Reply by marlen-subaru** [reply]
Thanks for your reply. I was surprised that they would answer me. I will definitely prepare all the information soon so that you can familiarize yourself with it. Thank you for your attention😊

**Comment by marlen-subaru** [comment]
Hi. I made a mistake in the library name, TFT_eSPI. Using it, I can output a picture (broadcast) from the esp32-cam camera onto the gc9a01 tft screen. But, I want to use LovyanGFX. My first error after initialization is: "Compilation error: 'LGFX' does not name a type". Help me figure out what's missing. And in which file, you will need to make changes to connect gc9a01 (these are pins, driver, screen resolution, etc.). The library is very large. Below is my sketch. It has the error described above. Thank you in advance. I will be waiting for an answer.

```cpp
#include <LovyanGFX.h>
#include <LovyanGFX.hpp>
#include "esp_camera.h"

#define CAM_PIN_PWDN    -1
#define CAM_PIN_RESET   -1
#define CAM_PIN_XCLK    21
#define CAM_PIN_SIOD   26
#define CAM_PIN_SIOC   27
#define CAM_PIN_D0     34
#define CAM_PIN_D1     35
#define CAM_PIN_D2     32
#define CAM_PIN_D3     33
#define CAM_PIN_D4     14
#define CAM_PIN_D5     13
#define CAM_PIN_D6     12
#define CAM_PIN_D7     15
#define CAM_PIN_VSYNC  25
#define CAM_PIN_HREF   23
#define CAM_PIN_PCLK   22

void setupCamera() {
    camera_config_t config;
    config.ledc_channel = LEDC_CHANNEL_0;
    config.ledc_timer = LEDC_TIMER_0;
    config.pin_d0 = CAM_PIN_D0;
    config.pin_d1 = CAM_PIN_D1;
    config.pin_d2 = CAM_PIN_D2;
    config.pin_d3 = CAM_PIN_D3;
    config.pin_d4 = CAM_PIN_D4;
    config.pin_d5 = CAM_PIN_D5;
    config.pin_d6 = CAM_PIN_D6;
    config.pin_d7 = CAM_PIN_D7;
    config.pin_xclk = CAM_PIN_XCLK;
    config.pin_pclk = CAM_PIN_PCLK;
    config.pin_vsync = CAM_PIN_VSYNC;
    config.pin_href = CAM_PIN_HREF;
    config.pin_pwdn = CAM_PIN_PWDN;
    config.pin_reset = CAM_PIN_RESET;
    config.xclk_freq_hz = 20000000;
    config.pixel_format = PIXFORMAT_JPEG;

    config.frame_size = FRAMESIZE_SVGA;
    config.jpeg_quality = 12;
    config.fb_count = 2;

    esp_err_t err = esp_camera_init(&config);
    if (err != ESP_OK) {
        Serial.println("Ошибка инициализации камеры");
        return;
    }
}


LGFX tft;  // error 1

void setup() {
    Serial.begin(115200);
    setupCamera();

    tft.init();   // error 2
    tft.setRotation(1);
    tft.fillScreen(TFT_BLACK);
}

void loop() {
    camera_fb_t *fb = esp_camera_fb_get();
    if (!fb) {
        Serial.println("Ошибка захвата изображения");
        return;
    }

 
    tft.drawJpg(fb->buf, fb->len, 0, 0);   // error 3
    esp_camera_fb_return(fb);

    delay(100); 
}
```

**Reply by tobozo** [reply]
the error is because it's missing the panel configuration, so your first goal is to create a configuration and get the panel to work alone.

see this example:

https://github.com/lovyan03/LovyanGFX/blob/master/examples/HowToUse/2_user_setting/2_user_setting.ino
